### ISSUE TITLE
IPN vs Code Base identifiers - two registers, same org

### ISSUE THREAD

**Issue by Julianlstar** [ThreeSixtyGiving/standard#195]
Are we OK to use IPN identifiers for locations? They are listed in the Index of Place Names in Great Britain 2016 by the ONS. The challenge is that IPN number isn’t the same as Geo Code. The IPN database does have geocodes in it but not the same Ward level ones. So Heatons South appears in the IPN database as IPN0032814 and in the Codebase as E05000792. Two different registers same organisation. The problem is that the locality names are listed in the IPN register with associated IPN numbers. Ultimately for mapping do we quantise the localities to ward level and then associate it with the ward geocode from the Code Base? I can see it would make a bit of a mess of mapping.

**Comment by Julianlstar** [comment]
Just been referred to this which is quite a good explainer of statistical geography in the U.K. https://theidpblog.files.wordpress.com/2014/08/hierarchical_representation_of_uk_statistical_geographies_july_2014.pdf

**Comment by timgdavies** [comment]
Quick clarification: 

* By 'same org' - do you mean the same publisher of these two lists (.e. ONS?)? Or are you trying to use these as identifiers for 'Heatons South' as an organisation? 

I've just [explored the geoportal copy of this data](http://geoportal.statistics.gov.uk/datasets/index-of-place-names-in-great-britain-july-2016/data) and as I understand:

* Some place names have a 'wd15cd' field, which is the code of the Ward in which the place belongs;
* A place name might refer to a place smaller than a ward, identical to a ward, or bigger than ward level;

If your goal is to identify wards, then I would suggest using Codebase.

If your goal is to be able to identify places distinct from wards, then using IPN **and** ward codes would potentially be the most useful. It is possible to include multiple geographic codes within a single 360 Giving file.

**Comment by Julianlstar** [comment]
I was a bit vague with the question. The org I was referring to was ONS

**Comment by Julianlstar** [comment]
@timgdavies Following on from your point about multiple geographic codes. Does this example pulled from MACC dataset illustrate your point. 
![screen shot 2017-07-05 at 14 01 48](https://user-images.githubusercontent.com/3984183/27865471-91d250b2-618a-11e7-88d0-dcc9af46c51d.png)

**Comment by timgdavies** [comment]
Yes. From this we can see one grant, being classified both by Clinical Commissioning Group geography and electoral wards. 

There are two ways in 360Giving to provide one-to-many relationships (One grant, many location classifications):

* Sub-tables, with one row per classification - this is the approach taken in the MACC file
* Repeated columns (with an index indicator) such that you can have (for example) one column that you use for ward code (with a hidden column stating the type for the preceding column) and one column for CCG code (with another hidden column to indicate it's type). 

The second approach is useful when you want to offer just two or three options for a single location per grants. There is more on spreadsheet design in the [flatten tool handbook](http://flatten-tool.readthedocs.io/en/latest/unflatten/#human-friendly-headings-using-a-json-schema-with-titles).

**Comment by morchickit** [comment]
We need to makes sure people are aware we allow multiple geographies for a grant.
